### PR TITLE
Change 'Fail' to 'Error' in formattedMessage designation (to clarify …

### PR DESCRIPTION
**Pull request by michaelcfanning**
…this matches the Result kind. Add a check DoNotModifyStackProtectionCookie to detect packed files. Add upx-packed test file. @lgolding @nguerrera @BillyONeal @bobschumaker @tikiwanMS @davidknise

### CODE REVIEW

File: src/BinSkim.Rules/CryptoErrors.cs [hunk 1]
```diff
@@ -68,10 +68,10 @@ public enum CryptoError : uint
         CERT_E_MALFORMED = 0x800B0108, // A certificate is missing or has an empty value for an important field, such as a subject or issuer name.  
         CERT_E_UNTRUSTEDROOT = 0x800B0109, // A certificate chain processed correctly, but terminated in a root certificate which is not trusted by the trust provider.  
         CERT_E_CHAINING = 0x800B010A, // An internal certificate chaining error has occurred.  
-        TRUST_E_FAIL = 0x800B010B, // Generic trust failure.  
+        TRUST_E_Error = 0x800B010B, // Generic trust failure.
```
Comment (michaelcfanning) [inline src/BinSkim.Rules/CryptoErrors.cs]: Fixed these bad renames in a second commit. refresh this PR to see.

File: src/BinSkim.Rules/DoNotModifyStackProtectionCookie.cs [hunk 2]
```diff
@@ -87,16 +88,23 @@ public override void Analyze(BinaryAnalyzerContext context)
...
                 {
                     return;
                 }
             }
-            else if (!Validate32BitImage(context))
+            catch (Exception)
             {
-                return;
+                Console.WriteLine();
             }
```
Comment (nguerrera) [inline src/BinSkim.Rules/DoNotModifyStackProtectionCookie.cs]: Why are all exceptions completely ignored here and why the Console.WriteLine()?

File: src/BinSkim.Rules/NativeInterop.cs [hunk 3]
```diff
@@ -26,10 +26,10 @@ public enum WinVerifyTrustErrors : uint
             CERT_E_MALFORMED = 0x800B0108,
             CERT_E_UNTRUSTEDROOT = 0x800B0109,
             CERT_E_CHAINING = 0x800B010A,
-            TRUST_E_FAIL = 0x800B010B,
+            TRUST_E_Error = 0x800B010B,
             CERT_E_REVOKED = 0x800B010C,
             CERT_E_UNTRUSTEDTESTROOT = 0x800B010D,
-            CERT_E_REVOCATION_FAILURE = 0x800B010E,
+            CERT_E_REVOCATION_ErrorURE = 0x800B010E,
```
Comment (nguerrera) [inline src/BinSkim.Rules/NativeInterop.cs]: Two more bad renames here. Looks like ea876669461091a63e8c3a70cfbd86309e2561b6 only fixed CryptoErrors.cs.

File: src/BinaryParsers/PortableExecutable/SafePointer.cs [hunk 4]
```diff
@@ -262,7 +262,7 @@ public bool HasData(int cBytes)
         private void TestPointerAndThrow()
         {
             if (!IsValid)
-                throw new Exception("Pointer is beyound the safe range");
+                throw new InvalidOperationException("Pointer is beyound the safe range.");
```
Comment (nguerrera) [inline src/BinaryParsers/PortableExecutable/SafePointer.cs]: Pre-existing, but s/beyound/beyond/

**Pull request merged**